aws/credentials/bearer/token.go

```diff
@@ -0,0 +1,49 @@
+package bearer
```

Comment:
to echo one of @aajtodd comments earlier, its good that you removed this out of the ssocreds package. but i think this tokenprovider should be out of the credentials package completely.

it should be `aws/bearer/token.go`. this is because a token provider is not necessarily a "credentials" based auth, its its own thing. we just happen to be using it as a utility mechanism: use the token provider as a means of getting an access token that we then send to the SSO service to get aws credentials.

for example, `StaticTokenProvider` has no business being under the `credentials` package